Donation-platform/src/main/java/com/google/servlets/TagQueryServlet.java

```diff
@@ -0,0 +1,59 @@
+// // Copyright 2019 Google LLC
+// //
+// // Licensed under the Apache License, Version 2.0 (the "License");
+// // you may not use this file except in compliance with the License.
+// // You may obtain a copy of the License at
+// //
+// //     https://www.apache.org/licenses/LICENSE-2.0
+// //
+// // Unless required by applicable law or agreed to in writing, software
+// // distributed under the License is distributed on an "AS IS" BASIS,
+// // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// // See the License for the specific language governing permissions and
+// // limitations under the License.
+
+package com.google.servlets;
+
+import com.google.FindTrendingCharities;
+
+import java.io.IOException;
+import java.util.Collection;
+import com.google.gson.Gson;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import com.google.model.Charity;
+import com.google.model.Tag;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+
+/** Servlet that handles requests for the trending page* */
+@WebServlet("/tag-query")
+public class TagQueryServlet extends HttpServlet {
```

Comment:
the name of this servlet is very confusing. documentation says trending page but name says tag. Should we rename it more meaningful? like TrendingPageScoreUpdateServlet

Reply:
changed it